jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/service/MigrateServiceTest.java

```diff
@@ -17,90 +17,93 @@
+import org.junit.jupiter.api.io.TempDir;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
-@RunWith(MockitoJUnitRunner.class)
-public class MigrateServiceTest {
-  @Mock
-  KitLogger logger;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.fail;
 
-  @Rule
-  public TemporaryFolder folder = new TemporaryFolder();
+class MigrateServiceTest {
+  private KitLogger logger;
+
+  @TempDir
+  Path folder;
+
+  @BeforeEach
+  public void setUp() {
+    logger = new KitLogger.SilentLogger();
+  }
 
   @Test
-  public void testPomPluginMigrationInBuild() throws Exception {
+  void testPomPluginMigrationInBuild() throws Exception {
     // Given
     File pomFile = copyToTempDirectory("/service/migrate/test-project/fabric8-pom.xml");
     // When
-    new MigrateService(folder.getRoot(), logger).migrate(
+    new MigrateService(folder.toFile(), logger).migrate(
         "org.eclipse.jkube", "kubernetes-maven-plugin", "1.0.0-SNAPSHOT");
     // Then
     assertExpectedDocument("/service/migrate/test-project/expected-pom.xml").accept(pomFile);
   }
 
   @Test
-  public void testPomPluginMigrationInProfile() throws Exception {
+  void testPomPluginMigrationInProfile() throws Exception {
     // Given
     File pomFile = copyToTempDirectory("/service/migrate/test-project-profile/fabric8-pom.xml");
     // When
-    new MigrateService(folder.getRoot(), logger).migrate(
+    new MigrateService(folder.toFile(), logger).migrate(
         "org.eclipse.jkube", "openshift-maven-plugin", "1.0.0-SNAPSHOT");
     // Then
     assertExpectedDocument("/service/migrate/test-project-profile/expected-pom.xml").accept(pomFile);
   }
 
   @Test
-  public void testProjectResourceFragmentDirectoryRename() throws Exception {
+  void testProjectResourceFragmentDirectoryRename() throws Exception {
     // Given
     copyToTempDirectory("/service/migrate/test-project/fabric8-pom.xml");
-    final File fabric8 = folder.newFolder("src", "main", "fabric8");
-    assertTrue(new File(fabric8, "file1").createNewFile());
+    final File fabric8 =
+            Files.createDirectories(folder.resolve("src").resolve("main").resolve("fabric8")).toFile();
+    assertThat(new File(fabric8, "file1").createNewFile()).isTrue();
     // When
-    new MigrateService(folder.getRoot(), logger)
+    new MigrateService(folder.toFile(), logger)
         .migrate("org.eclipse.jkube", "openshift-maven-plugin", "1.0.0-SNAPSHOT");
     // Then
-    assertFalse(fabric8.exists());
-    assertTrue(new File(folder.getRoot(), "src/main/jkube").exists());
-    assertTrue(new File(folder.getRoot(), "src/main/jkube/file1").exists());
+    assertThat(fabric8).doesNotExist();
+    assertThat(new File(folder.toFile(), "src/main/jkube")).exists();
+    assertThat(new File(folder.toFile(), "src/main/jkube/file1")).exists();
```

Comment:
This would be a good time to refactor this assertion so that it use cross-os file separators

Reply:
Do you mean this?
```java
assertThat(new File(folder, "src" + fileSeparator + "main" + fileSeparator + "jkube")).exists();
assertThat(new File(folder, "src" + fileSeparator + "main" + fileSeparator + "jkube" + fileSeparator + "file1")).exists();
```

Reply:
Many options available.
- What you suggested
- Using Path assertions:
  ```java

    assertThat(folder.resolve("src").resolve("main").resolve("jkube"))
        .exists().isDirectory();
    assertThat(folder.resolve("src").resolve("main").resolve("jkube").resolve("file1"))
        .exists().isRegularFile();
  ```
- Using a single Path assertions
  ```java
    assertThat(folder)
        .extracting(p -> p.resolve("src").resolve("main").resolve("jkube"))
        .asInstanceOf(InstanceOfAssertFactories.PATH)
        .exists().isDirectory()
        .extracting(p -> p.resolve("file1"))
        .asInstanceOf(InstanceOfAssertFactories.PATH)
        .exists().isRegularFile();
   ```
- Anything else that you can think of...

Last but not least, you could use our `FileAssertions.assertThat(folder.toFile()).fileTree()`, but that'd require some additional changes to our utility file so we consider paths in an OS agnostic way. (Maybe we can do this in the **future**, I think there was some work done in this in #1416. I'll probably cherry-pick some changes so we can start integrating those improvements)

Reply:
I think single Path assertions would be nice